### ISSUE TITLE
Pipewire breaks infrequent writers between pipewire 0.3.59 and 0.3.63

### ISSUE THREAD

**Issue by hartmans** [espeak-ng/pcaudiolib#23]
Hi.  In #12  it's pointed out that explicit pipewire support is not a high priority because pipewire's pulse support is generally good enough.
Unfortunately, I've found that somewhere between pipewire 0.59 and 0.63, the way emacspeak uses libespeak-ng got broken.
I care about this a lot because emacspeak is critical for me for accessibility, and without it I don't have great ways to use terminal applications.
I've reduced things to a much simpler test case:
```c
#include <string.h>
#include <assert.h>
#include <unistd.h>
#include <espeak-ng/speak_lib.h>
const char * a_str = "Begin";
const char * b_str = "end now";
int main() {
  unsigned int unique_identifier = 0;
  espeak_Initialize(AUDIO_OUTPUT_PLAYBACK, 512, NULL, 0);
  assert(espeak_SetParameter(espeakRATE, 550, 0) == EE_OK);
  espeak_Synth(a_str, strlen(a_str)+1, 0, POS_CHARACTER, 0,
                     espeakCHARS_UTF8 , &unique_identifier, NULL);
  sleep(3);
  espeak_Synth(b_str, strlen(b_str), 0, POS_CHARACTER, 0,
                     espeakCHARS_UTF8 , &unique_identifier, NULL);
  sleep(3);
}
```
Obviously that should be linked against `-lespeak-ng`
Under pipewire 0.59, I hear both 'begin' and 'end now'.
Under pipewire 0.63, I only hear 'begin'.
What I think is happening is that pipewire-pulse or pipewire has changed the handling of streams that underrun.

Emacspeak expects that it can generate speech whenever it likes and doesn't need to care if the audio stream underruns.
I'm reporting this against the audio library rather than espeak-ng as I suspect that's probably where the fix will end up being.
(Unless it's on the pipewire side).
I'd love to find a solution to this so we can get it into Debian for the bookworm release.
CC: @sthibaul

**Comment by hartmans** [comment]
Oops.  I had my pipewire versions abreviated.  The good version is 0.3.59 and the bad version is 0.3.63.  I don't appear to have permission to edit the issue title.

**Comment by sthibaul** [comment]
One thing I know they changed is to auto-suspend the device after a few seconds when the audio underruns, to save battery. See https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2839 . The default timeout is 5s, however, so that would not really fit. But I guess it could be an interesting starting point, and discussing with pipewire authors on their gitlab could be fruitful.

**Comment by daissi** [comment]
Forwarded to pipewire: https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2928

**Comment by hartmans** [comment]
Fixed in pipewire commit 624e265fd6e5027e11027822eb943c533feab85f
